src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts

```diff
@@ -70,34 +81,46 @@ export default class UBAFeeSpokeCalculator {
     // If the last validated running balance is undefined, we need to compute the running balance from scratch
     // This is the case when the UBA Fee Calculator is first initialized or run on a range
     // that we haven't computed the running balance for yet
-    const historicalResult = this.recentRequestFlow.slice(startIdx, endIdx).reduce((acc, flow) => {
-      let resultantValue = acc;
-      if (isUbaInflow(flow)) {
-        resultantValue = acc.add(toBN(flow.amount));
-      } else if (isUbaOutflow(flow)) {
-        resultantValue = acc.sub(toBN(flow.amount));
+    const historicalResult: RunningBalanceUnion = this.recentRequestFlow.slice(startIdx, endIdx).reduce(
+      (acc, flow) => {
+        const resultant: RunningBalanceUnion = { ...acc };
+
+        if (isUbaInflow(flow)) {
+          resultant.runningBalance = acc.runningBalance.add(flow.amount);
+          resultant.incentivePoolBalance = acc.incentivePoolBalance.add(flow.amount);
```

Comment:
Mark a placeholder here for computing the actual incentive delta?

Reply:
Good call! Updated here: https://github.com/across-protocol/sdk-v2/pull/212/commits/2e810041719fce1711ba1f910b6fd66cd751b300